sql/src/main/java/org/apache/druid/sql/calcite/schema/DruidSchema.java

```diff
@@ -351,7 +352,8 @@ public void awaitInitialization() throws InterruptedException
     return builder.build();
   }
 
-  private void addSegment(final DruidServerMetadata server, final DataSegment segment)
+  @VisibleForTesting
+  protected void addSegment(final DruidServerMetadata server, final DataSegment segment)
```

Comment:
Would this be better as package private instead of protected?  Since there's no subclasses in other packages, is there a need to be protected?